pennylane/plugins/default_qubit.py

```diff
@@ -453,6 +519,18 @@ def _get_operator_matrix(self, operation, par):
             return A
         return A(*par)
 
+    def _get_tensor_operator_matrix(self, tensor):
+        """Get the operator matrix for a given tensor of operations.
+
+        Args:
+          tensor (:class:`Tensor`): A tensor of operations
+          par (tuple[float]): parameter values
+        Returns:
+          array: matrix representation.
+        """
+        ops = [self._get_operator_matrix(op.name, op.parameters) for op in tensor.ops]
+        return np.kron(*ops)
```

Comment:
:+1:

Reply:
If I'm reading right, this will always measure by passing the full dense matrix. @josh146 is there a case where it would make sense to pass the individual matrices of the tensor product to backend, without combining them?

Reply:
Currently, all PL does is pass the _names_ of the observable in the tensor product to the plugin, not even the matrices, so it's completely up to the plugin how it is implemented. I think this is the best approach due to the differences across frameworks.

For this particular case here, there are more efficient approaches we can make use of inside `default.qubit`. This could include performing single-qubit basis rotations, and simply measuring in the Z^{\otimes N} basis.